From 413d1008d96a44cba5bf82030f7e95450ae82554 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Johannes=20L=C3=B6thberg?= <johannes@kyriasis.com>
Date: Sat, 15 May 2021 12:25:10 +0200
Subject: [PATCH] Revert "Close #7996: manpage: Make a section directory on
 build manpage by default"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This reverts commit a843fd3db8b588be93b576049a8a291bc4b3ec9d.

Signed-off-by: Johannes Löthberg <johannes@kyriasis.com>
---
 CHANGES                     |  2 --
 doc/usage/configuration.rst |  4 +---
 sphinx/builders/manpage.py  |  2 +-
 tests/test_build_manpage.py | 12 ++++++------
 tests/test_smartquotes.py   |  4 ++--
 5 files changed, 10 insertions(+), 14 deletions(-)

diff --git a/CHANGES b/CHANGES
index b7e4c7f0e..7ca08a6ac 100644
--- a/CHANGES
+++ b/CHANGES
@@ -50,8 +50,6 @@ Incompatible changes
 * #7784: i18n: The msgid for alt text of image is changed
 * #5560: napoleon: :confval:`napoleon_use_param` also affect "other parameters"
   section
-* #7996: manpage: Make a section directory on build manpage by default (see
-  :confval:`man_make_section_directory`)
 * #7849: html: Change the default setting of
   :confval:`html_codeblock_linenos_style` to ``'inline'``
 * #8380: html search: search results are wrapped with ``<p>`` instead of
diff --git a/doc/usage/configuration.rst b/doc/usage/configuration.rst
index 7b7a0a1d4..3bc402c9a 100644
--- a/doc/usage/configuration.rst
+++ b/doc/usage/configuration.rst
@@ -2332,12 +2332,10 @@ These options influence manual page output.
 
 .. confval:: man_make_section_directory
 
-   If true, make a section directory on build man page.  Default is True.
+   If true, make a section directory on build man page.  Default is False.
 
    .. versionadded:: 3.3
-   .. versionchanged:: 4.0
 
-      The default is changed to ``False`` from ``True``.
 
 .. _texinfo-options:
 
diff --git a/sphinx/builders/manpage.py b/sphinx/builders/manpage.py
index b993a2df4..532d2b8fe 100644
--- a/sphinx/builders/manpage.py
+++ b/sphinx/builders/manpage.py
@@ -118,7 +118,7 @@ def setup(app: Sphinx) -> Dict[str, Any]:
 
     app.add_config_value('man_pages', default_man_pages, None)
     app.add_config_value('man_show_urls', False, None)
-    app.add_config_value('man_make_section_directory', True, None)
+    app.add_config_value('man_make_section_directory', False, None)
 
     return {
         'version': 'builtin',
diff --git a/tests/test_build_manpage.py b/tests/test_build_manpage.py
index 2affb8ba3..a017abc69 100644
--- a/tests/test_build_manpage.py
+++ b/tests/test_build_manpage.py
@@ -17,9 +17,9 @@ from sphinx.config import Config
 @pytest.mark.sphinx('man')
 def test_all(app, status, warning):
     app.builder.build_all()
-    assert (app.outdir / '1' / 'sphinxtests.1').exists()
+    assert (app.outdir / 'sphinxtests.1').exists()
 
-    content = (app.outdir / '1' / 'sphinxtests.1').read_text()
+    content = (app.outdir / 'sphinxtests.1').read_text()
     assert r'\fBprint \fP\fIi\fP\fB\en\fP' in content
     assert r'\fBmanpage\en\fP' in content
 
@@ -31,16 +31,16 @@ def test_all(app, status, warning):
 
 
 @pytest.mark.sphinx('man', testroot='basic',
-                    confoverrides={'man_make_section_directory': False})
+                    confoverrides={'man_make_section_directory': True})
 def test_man_make_section_directory(app, status, warning):
     app.build()
-    assert (app.outdir / 'python.1').exists()
+    assert (app.outdir / '1' / 'python.1').exists()
 
 
 @pytest.mark.sphinx('man', testroot='directive-code')
 def test_captioned_code_block(app, status, warning):
     app.builder.build_all()
-    content = (app.outdir / '1' / 'python.1').read_text()
+    content = (app.outdir / 'python.1').read_text()
 
     assert ('.sp\n'
             'caption \\fItest\\fP rb\n'
@@ -71,5 +71,5 @@ def test_default_man_pages():
 @pytest.mark.sphinx('man', testroot='markup-rubric')
 def test_rubric(app, status, warning):
     app.build()
-    content = (app.outdir / '1' / 'python.1').read_text()
+    content = (app.outdir / 'python.1').read_text()
     assert 'This is a rubric\n' in content
diff --git a/tests/test_smartquotes.py b/tests/test_smartquotes.py
index a8d4b1ed6..4879dedc6 100644
--- a/tests/test_smartquotes.py
+++ b/tests/test_smartquotes.py
@@ -31,7 +31,7 @@ def test_text_builder(app, status, warning):
 def test_man_builder(app, status, warning):
     app.build()
 
-    content = (app.outdir / '1' / 'python.1').read_text()
+    content = (app.outdir / 'python.1').read_text()
     assert '\\-\\- "Sphinx" is a tool that makes it easy ...' in content
 
 
@@ -84,5 +84,5 @@ def test_smartquotes_excludes_language(app, status, warning):
 def test_smartquotes_excludes_builders(app, status, warning):
     app.build()
 
-    content = (app.outdir / '1' / 'python.1').read_text()
+    content = (app.outdir / 'python.1').read_text()
     assert '– “Sphinx” is a tool that makes it easy …' in content
-- 
2.31.1

